### PR TITLE
[doc][build] various changes to connect local doc build together

### PR DESCRIPTION
**Pull request by khluu**
- Include dependency changes from https://github.com/ray-project/ray/pull/46867
- Since the bucket allows public use, it doesn't work with my AWS credential anymore but adding `UNSIGNED` (which means it's somewhat anonymous call? equivalent to `--no-sign-request` when using AWS CLI) works
- Change the command to get latest commit to `git merge-base HEAD origin/master` since it seems to be more accurate.
- Add `local` option to Makefile which does incremental build without `-a -E` flags
- Change `main-sidebar` to use `generate_toctree` instead of `cached_toctree`

### CODE REVIEW

**Comment by khluu** [comment]
doc built in 5s on my mac machine :0

**Comment by can-anyscale** [comment]
impossible!

File: ci/ray_ci/doc/cmd_update_cache_env.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def update_file_timestamp(ray_dir: str) -> None:
 def main(ray_dir: str) -> None:
     pending_files = list_pending_files(ray_dir)
     update_environment_pickle(ray_dir, pending_files)
-    # update_file_timestamp(ray_dir)
+    update_file_timestamp(ray_dir)
```
Comment (can-anyscale) [inline ci/ray_ci/doc/cmd_update_cache_env.py:87]: can we add a TODO here to only update this for untracked files?

**Review by can-anyscale — APPROVED** [review]

File: ci/ray_ci/doc/cmd_update_cache_env.py (hunk 1 above)
Comment (khluu) [reply ci/ray_ci/doc/cmd_update_cache_env.py:87]: oh I added a TODO before the method def

**Pull request merged**